sdk/purview/purview-workflow-rest/samples-dev/workflowRunSample.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import createPurviewWorkflowClient, {
+  CancelWorkflowRunParameters,
+  ListWorkflowRunsParameters,
+  paginate,
+} from "@azure-rest/purview-workflow";
+import { UsernamePasswordCredential } from "@azure/identity";
+import * as dotenv from "dotenv";
+
+dotenv.config();
+
+/**
+ * This sample demonstrates how to List workflow runs.
+ *
+ * @summary List workflow runs.
+ * x-ms-original-file: specification/purview/data-plane/Azure.Analytics.Purview.Workflow/preview/2022-05-01-preview/examples/ListWorkflowRuns.json
+ */
+async function workflowRunsList() {
+  const endpoint = process.env["ENDPOINT"] || "";
+  const tenantId = process.env["TENANTID"] || "";
+  const clientId = process.env["CLIENTID"] || "";
+  const username = process.env["USERNAME"] || "";
+  const password = process.env["PASSWORD"] || "";
+  const credential = new UsernamePasswordCredential(tenantId, clientId, username, password);
+  const client = createPurviewWorkflowClient(endpoint, credential);
+  const options: ListWorkflowRunsParameters = {
+    queryParameters: {
+      timeWindow: "30d",
+      orderby: "startTime desc",
+      maxpagesize: 1000,
+    },
+  };
+  const initialResponse = await client.path("/workflowruns").get(options);
+  const pageData = paginate(client, initialResponse);
+  const result = [];
+  for await (const item of pageData) {
+    result.push(item);
+  }
+  console.log(result);
+}
+
+/**
+ * This sample demonstrates how to Get a workflow run.
+ *
+ * @summary Get a workflow run.
+ * x-ms-original-file: specification/purview/data-plane/Azure.Analytics.Purview.Workflow/preview/2022-05-01-preview/examples/GetWorkflowRun.json
+ */
+async function workflowRunGet() {
+  const endpoint = process.env["ENDPOINT"] || "";
+  const tenantId = process.env["TENANTID"] || "";
+  const clientId = process.env["CLIENTID"] || "";
+  const username = process.env["USERNAME"] || "";
+  const password = process.env["PASSWORD"] || "";
+  const credential = new UsernamePasswordCredential(tenantId, clientId, username, password);
+  const client = createPurviewWorkflowClient(endpoint, credential);
+  const workflowRunId = "04334a74-3400-11ed-a261-0242ac120002";
```

Comment:
Hardcoding an ID makes it hard for customers to run the sample. How can a customer create a workflow run? Also, is it a long-running operation? Did you consider creating a poller for it?

Reply:
It is due to workflow run is created in Workflow automatically, it is not created by user, so user could only get the workflow run id from list workflow run api, I explained in here on how to get the workflow run id : https://github.com/Azure/azure-sdk-for-js/blob/71bdfe63f8576e0b32305f29f970a2e0204a77ad/sdk/purview/purview-workflow-rest/samples-dev/workflowRunSample.ts#L90

Workflow run is not a long-running operation.

Reply:
Can you please add the same comment about how to get the ID here and also just set a "<WorkflowRunId>". We expect customers to copy/paste these samples and hardcoding the GUID here will make people think it should be also copied.

Also I would like to see in these samples how to use the result. What do you expect your customers to do with your result?

At the very least

```ts
if(isUnexpected(result)){
   throw result.body.error;
}

console.log(`The run has result is ${result.body.someInterestingData}`)

```

This applies for all samples

Reply:
Fixed, thanks, I removed the hard code guid in sample and add more comment in samples to explain how to use APIs in samples.